### PR TITLE
remove null to ExecutionContext.global

### PR DESCRIPTION
**Pull request by YuMuuu**
I wondered use a null value to executionContext.global 
argument.
And match null for `impl.ExecutionContextImpl.fromExecutor(null: Executor)`

I thought it could be improved.

### CODE REVIEW

File: src/library/scala/concurrent/impl/ExecutionContextImpl.scala [hunk 1]
```diff
@@ -116,29 +116,24 @@ private[concurrent] object ExecutionContextImpl {
...
-            final override def submit[T](runnable: Runnable, t: T) = asExecutorService.submit(runnable, t)
-            final override def submit(runnable: Runnable) = asExecutorService.submit(runnable)
-            final override def invokeAll[T](callables: Collection[_ <: Callable[T]]) = asExecutorService.invokeAll(callables)
-            final override def invokeAll[T](callables: Collection[_ <: Callable[T]], l: Long, timeUnit: TimeUnit) = asExecutorService.invokeAll(callables, l, timeUnit)
-            final override def invokeAny[T](callables: Collection[_ <: Callable[T]]) = asExecutorService.invokeAny(callables)
-            final override def invokeAny[T](callables: Collection[_ <: Callable[T]], l: Long, timeUnit: TimeUnit) = asExecutorService.invokeAny(callables, l, timeUnit)
-          }
-        }
+    ExecutionContextExecutorService =
+      new ExecutionContextImpl(es, reporter) with ExecutionContextExecutorService {
```
Comment (viktorklang) [inline src/library/scala/concurrent/impl/ExecutionContextImpl.scala:129]: Might be worth throwing an IllegalArgumentException if `es` is `null` here.

File: src/library/scala/concurrent/impl/ExecutionContextImpl.scala [hunk 2]
```diff
@@ -116,29 +116,24 @@ private[concurrent] object ExecutionContextImpl {
   }
 
   def fromExecutor(e: Executor, reporter: Throwable => Unit = ExecutionContext.defaultReporter): ExecutionContextExecutor =
-    e match {
-      case null => createDefaultExecutorService(reporter)
-      case some => new ExecutionContextImpl(some, reporter)
-    }
+    new ExecutionContextImpl(e, reporter)
```
Comment (viktorklang) [inline src/library/scala/concurrent/impl/ExecutionContextImpl.scala]: Might be worth throwing an IllegalArgumentException if `es` is `null` here.

**Review by viktorklang — CHANGES_REQUESTED** [review]
LGTM after comments

**Comment by YuMuuu** [comment]
@viktorklang 
thank you for reply.
I fix failed tests (sry T_T) and correct the review content.

**Comment by som-snytt** [comment]
I don't have an actual opinion, but I was surprised by Viktor's OK, because of compatibility. Previously, I have wished for an easy API to "get the default context." I remember thinking that "give me null" is suboptimal. Anything that requires saying `concurrent.impl` is no longer API. I'm too lazy to figure out if the change means "I used to supply a null, but now I must reference impl.something."

**Comment by viktorklang** [comment]
@som-snytt My reason for the OK was that `object ExecutionContextImpl` is `private[concurrent]`.

**Comment by YuMuuu** [comment]
The job exceeded the maximum time limit for jobs, and has been terminated. 🤔

File: test/files/jvm/future-spec/FutureTests.scala [hunk 3]
```diff
@@ -230,7 +230,7 @@ class FutureTests extends MinimalScalaTest {
     import ExecutionContext.Implicits._
     "report uncaught exceptions" in {
       val p = Promise[Throwable]()
-      val ec: ExecutionContextExecutorService = ExecutionContext.fromExecutorService(null, p.trySuccess(_))
+      val ec: ExecutionContextExecutorService = ExecutionContext.global
```
Comment (viktorklang) [inline test/files/jvm/future-spec/FutureTests.scala:233]: You can't use global here—the test shuts down the newly created `ec`.

Comment (viktorklang) [reply test/files/jvm/future-spec/FutureTests.scala:233]: See line 239

**Comment by SethTisue** [comment]
@YuMuuu interested in returning to this?

**Comment by SethTisue** [comment]
@YuMuuu second ping, interested in returning to this?

**Comment by YuMuuu** [comment]
@SethTisue Sorry. I forgot that I was an softrware engineer. I do it.

**Comment by SethTisue** [comment]
closing for inactivity, but there's no intent to discourage from progressing, we can reopen anytime

**Pull request closed without merging**